### DISCUSSION TITLE
Is it possible to use third party library in worklet?

### DISCUSSION THREAD

**Discussion by marcato15** [software-mansion/react-native-reanimated#3330] [Q&A] [answered]
I have a third party library I'd like to be able to use inside a worklet. What I'm not sure about is when a function is exposed does every function inside that function (like in a library) need to be defined as a worklet? Would it even be possible to run a third party library inside a worklet if that is the case, without majorly refactoring the code base?

**Comment by nandorojo** [comment, marked as answer]
only if you use `runOnJs`. you can't run third-party non-worklets in a worklet

**Comment by sumitmanral** [comment]
can we await on runOnJS ?

**Comment by sumitmanral** [comment]
I am using 3rd party lib - numbro but I need to await on the result . How can we achieve it?

**Reply by eduance** [reply]
You can useSharedValue to store the result, this is available on both the JS and UI thread.

```
import { useSharedValue } from 'react-native-reanimated';

function App() {
  const width = useSharedValue(100);

  function myWorklet() {
    'worklet';
    width.value += 50;
  }

  useEffect(() => {
    console.log(width.value); // available on both JS and UI thread
  }, []);
}
```
